proto/config/config.proto

```diff
@@ -0,0 +1,95 @@
+syntax = "proto3";
+
+package linkerd2.config;
+
+option go_package = "github.com/linkerd/linkerd2/controller/gen/config";
+
+// ConfigStore provides an abstraction over the underlying configuration storge.
+service ConfigStore {
+  // Return the JSON-serialized global config.
+  rpc GetGlobalConfig(GetParam) returns (GlobalConfig) {}
+
+  // Return the JSON-serialized proxy config.
+  rpc GetProxyConfig(GetParam) returns (ProxyConfig) {}
+
+  // Set the global config. It overwrites the existing value.
+  rpc SetGlobalConfig(GlobalConfig) returns (Result) {}
+
+  // Set the proxy config. It overwrites the existing value.
+  rpc SetProxyConfig(ProxyConfig) returns (Result) {}
+}
+
+message GlobalConfig {
+  string api_addr = 1;
+  string linkerd_namespace = 2;
+  string linkerd_version = 3;
+  string cluster_domain = 4;
+  bool no_init_container = 5;
+  bool ha = 6;
+  bool auto_inject = 7;
+}
+
+message ProxyConfig {
+  Image proxy_image = 1;
+  Image proxy_init_image = 2;
+
+  Port api_port = 3;
+  Port control_port = 4;
+  repeated Port ignore_inbound_ports = 5;
+  repeated Port ignore_outbound_ports = 6;
+  Port inbound_port = 7;
+  Port metrics_port = 8;
+  Port outbound_port = 9;
+
+  Resource resource = 10;
```

Comment:
Right now `linkerd inject` lets you set resource requests, but there's also an open issue (#2029) for setting resource limits. It would be good to be more specific here about what this field represents. How about:

```
  ResourceRequirements resource_requests = 10;
```

(I recommend using `ResourceRequirements` to match the Kubernetes API)